pkg/renderer/sgml/renderer.go

```diff
@@ -0,0 +1,360 @@
+package sgml
+
+import (
+	"bytes"
+	htmltemplate "html/template"
+	"io"
+	"strings"
+	texttemplate "text/template"
+
+	"github.com/bytesparadise/libasciidoc/pkg/configuration"
+	"github.com/bytesparadise/libasciidoc/pkg/types"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+// Renderer implements the backend render interface by using sgml.
+type Renderer interface {
```

Comment:
in order to keep the code as simple as possible, could we avoid introducing this interface for now? I understand it may be needed in the future, but only at this point should we have it, when we know exactly what we need.

Reply:
So looking at this, the html5 library needs to instantiate an object (an sgmlRenderer) with specific templates.  The xhtml5 library will do something similar.

So we need an instance to carry this.  At the same time, I would prefer to avoid exposing the details of the internals of the sgmlRenderer to consumers.  That way leads to API promises, and broken promises and broken hearts later, when we need to change things.

So I'd strongly prefer to keep this as an interface for now.

This can also help with writing test cases later, because stub implementations can be written to conform to the interface.  (For example, you might want to write a test Renderer that can be used to validate the document structure passed in by the parser.)

Interfaces are your friend, and one of the greatest aspects of go.  Don't fear them. :-)

Reply:
For the record, I don't fear interfaces! I just don't want to introduce unnecessary artifacts that make the code more complex to maintain.